src/webpack/index.js

```diff
@@ -39,8 +67,60 @@ class URLImportPlugin {
   }
 
   apply(compiler) {
-    const moduleAssets = {};
+    const options = compiler?.options;
+    const chunkSplitting = options?.optimization?.splitChunks?.cacheGroups || {};
+    chunkSplitting.interleave = {
+      test(module) {
```

Comment:
@theKashey this was pretty neat. `test` provides the whole module, so im able to actually look up the module source and see if its an externalized chunk before placing it into a cacheGroup, i then can give the files a name. This allows me to code split anything by just moving them into another cache group using splitChunks and testing that the module meets the requirement... Automated code-splitting regardless of sync or async. Need to see how it holds up against tree shaking